x-pack/plugins/enterprise_search/public/applications/app_search/components/engine/engine_nav.test.tsx

```diff
@@ -32,106 +82,153 @@ describe('EngineRouter', () => {
 });
 
 describe('EngineNav', () => {
+  const values = { myRole: {}, engineName: 'some-engine', dataLoading: false, engine: {} };
+
   beforeEach(() => {
-    (useParams as jest.Mock).mockReturnValue({ engineName: 'some-engine' });
+    setMockValues(values);
+  });
+
+  it('does not render if async data is still loading', () => {
+    setMockValues({ ...values, dataLoading: true });
+    const wrapper = shallow(<EngineNav />);
+    expect(wrapper.isEmptyRender()).toBe(true);
   });
 
   it('does not render without an engine name', () => {
-    setMockValues({ myRole: {} });
-    (useParams as jest.Mock).mockReturnValue({ engineName: '' });
+    setMockValues({ ...values, engineName: '' });
     const wrapper = shallow(<EngineNav />);
     expect(wrapper.isEmptyRender()).toBe(true);
   });
 
-  it('renders an engine label', () => {
-    setMockValues({ myRole: {} });
-    const wrapper = mount(<EngineNav />);
+  it('renders an engine label and badges', () => {
+    setMockValues({ ...values, isSampleEngine: false, isMetaEngine: false });
+    const wrapper = shallow(<EngineNav />);
+    const label = wrapper.find('[data-test-subj="EngineLabel"]').find('.eui-textTruncate');
+
+    expect(label.text()).toEqual('SOME-ENGINE');
+    expect(wrapper.find(EuiBadge)).toHaveLength(0);
 
-    const label = wrapper.find('[data-test-subj="EngineLabel"]').last();
-    expect(label.text()).toEqual(expect.stringContaining('SOME-ENGINE'));
+    setMockValues({ ...values, isSampleEngine: true });
+    wrapper.setProps({}); // Re-render
```

Comment:
I have a PR coming immediately after this one that creates a new `rerender(wrapper);` test helper so we don't need all these clarification comments.